app/controllers/concerns/metrics_concern.rb

```diff
@@ -0,0 +1,96 @@
+# frozen_string_literal: true
+
+# Methods for logging usage metrics based on requests for files
+module MetricsConcern
+  def track_download(druid, file: nil)
+    return unless enabled?
+
+    ensure_visit
+    properties = { druid:, file: }.compact
+    metrics_service.track_event(event_data('download', properties))
+  end
+
+  private
+
+  # We're responsible for ensuring that every event is tied to a visit
```

Comment:
Without this, we aren't able to group events by session. The current specification for the metrics feature is that we want to show "unique" metrics in the same way that Zenodo does – i.e. grouping events by session, so that viewing something repeatedly in a short timeframe gets collapsed into one view. The Visit is the thing we group by to do this.

Visits also host all of the information about the user and their browser. Our requirements don't currently call for using this information, but it seems prudent to collect at least basic information in the event that we later want to examine trends.